contracts/sol6/katana/KatanaFeeHandler.sol

```diff
@@ -0,0 +1,459 @@
+        require(block.number > lastBurnBlock + burnBlockInterval, "wait more blocks to burn");
+
+        // update last burn block number
+        lastBurnBlock = block.number;
+
+        // Get amount to burn, if greater than weiToBurn, burn only weiToBurn per function call.
+        uint256 balance = address(this).balance;
+
+        // redundant check, can't happen
+        assert(balance >= totalPayoutBalance);
+        uint256 srcAmount = balance.sub(totalPayoutBalance);
+        srcAmount = srcAmount > weiToBurn ? weiToBurn : srcAmount;
+
+        // Get rate
+        uint256 kyberEthKncRate =
+            kyberProxy.getExpectedRateAfterFee(ETH_TOKEN_ADDRESS, knc, srcAmount, 0, "");
+        validateEthToKncRateToBurn(kyberEthKncRate);
+
+        // Buy some knc and burn
+        kncBurnAmount = kyberProxy.swapEtherToToken{value: srcAmount}(knc, kyberEthKncRate);
+
+        require(IBurnableToken(address(knc)).burn(kncBurnAmount), "knc burn failed");
+
+        emit KncBurned(kncBurnAmount, ETH_TOKEN_ADDRESS, srcAmount);
+        return kncBurnAmount;
+    }
+
+    /// @notice should be called off chain
+    /// @dev returns list of sanity rate contracts
+    /// @dev index 0 is currently used contract address, indexes > 0 are older versions
+    function getSanityRateContracts() external view returns (ISanityRate[] memory sanityRates) {
+        sanityRates = sanityRateContract;
+    }
+
+    /// @dev return latest knc/eth rate from sanity rate contract
+    function getLatestSanityRate() external view returns (uint256 kncToEthSanityRate) {
+        if (sanityRateContract.length > 0 && sanityRateContract[0] != ISanityRate(0)) {
+            kncToEthSanityRate = sanityRateContract[0].latestAnswer();
+        } else {
+            kncToEthSanityRate = 0;
+        }
+    }
+
+    function readBRRData() external view returns (uint256 rewardBps, uint256 rebateBps) {
+        rewardBps = uint256(brrData.rewardBps);
+        rebateBps = uint256(brrData.rebateBps);
+    }
+
+    function updateRebateValues(
+        uint256 rebateWei,
+        address[] memory rebateWallets,
+        uint256[] memory rebateBpsPerWallet
+    ) internal returns (uint256 totalRebatePaidWei) {
+        uint256 totalRebateBps;
+        uint256 walletRebateWei;
+
+        for (uint256 i = 0; i < rebateWallets.length; i++) {
+            require(rebateWallets[i] != address(0), "rebate wallet address 0");
+
+            walletRebateWei = rebateWei.mul(rebateBpsPerWallet[i]).div(BPS);
+            rebatePerWallet[rebateWallets[i]] = rebatePerWallet[rebateWallets[i]].add(
+                walletRebateWei
+            );
+
+            // a few wei could be left out due to rounding down. so count only paid wei
+            totalRebatePaidWei = totalRebatePaidWei.add(walletRebateWei);
+            totalRebateBps = totalRebateBps.add(rebateBpsPerWallet[i]);
+        }
+        require(totalRebateBps <= BPS, "rebates more then 100%");
```

Comment:
more than 100%

Reply:
fixed in https://github.com/KyberNetwork/kyber_protocol_sc/pull/11/commits/079ca9b8b0ae543679c04fc24001f7d3fe4ce5cd